app/services/course_validations.rb

```diff
@@ -19,8 +19,8 @@ def identical_to_existing_course?
   end
 
   def course_already_exists_on_application?
-    application_choices = application_choice.self_and_siblings - [application_choice].reject { |choice| ApplicationStateChange::UNSUCCESSFUL_STATES.include?(choice) }
```

Comment:
I found this piece of code hard to read.

It removes the current application choice from consideration when checking for duplicate course selections if the choice being validated is in an unsuccessful status.

This never happens, because `ApplicationStateChange::UNSUCCESSFUL_STATES.include?(choice)` is passing an AR object in and not a symbol representing the status.